src/purchase/views/purchase_view.py

```diff
@@ -79,6 +79,20 @@ def create(self, request):
         with transaction.atomic():
             user = User.objects.select_for_update().get(id=user.id)
 
+            # Check if a purchase was already created for this client ID. If a purchase exists
+            # the user probably accidentally double clicked submit and we should return an error.
+            purchase = Purchase.objects.filter(
```

Comment:
There may be some additional FE logic that does not pass a `client_id`. I wonder if it will throw an error. I think this is okay - I can update all spots to ensure `client_id` is passed